src/effects/VST/VSTEffect.cpp

```diff
@@ -1911,22 +1911,35 @@ std::vector<int> VSTEffect::GetEffectIDs()
    return effectIDs;
 }
 
+
+VstPatchChunkInfo VSTEffectWrapper::GetChunkInfo() const
+{
+   VstPatchChunkInfo info = { 1, mAEffect->uniqueID, mAEffect->version, mAEffect->numParams, "" };
+   return info;
+}
+
+bool VSTEffectWrapper::IsCompatible(const VstPatchChunkInfo& info) const
+{
+   return  (info.pluginUniqueID == mAEffect->uniqueID) &&
+           (info.pluginVersion  == mAEffect->version) &&
+           (info.numElements    == mAEffect->numParams);
+}
```

Comment:
The commit adding these two functions, "Rewrite LoadParameters not using the handles", is just hiding mention of mAeffect in LoadParameters while not removing dependency on its state.  It's not what I meant.

However, I must read and understand all that you commented in "Rewritten LoadSettings / SaveSettings".

Actually eliminating use of mAEffect might not be possible, as I think you are saying:  instead, this handle is really needed as a "scratch" as with AudioUnits.

Reply:
Let's try to see it from this angle: what LoadParameters has to do, is to read some stuff from the config, set it to a plugin handle (done in the call to callSetChunkB) and fetch the settings from the plugin handle. This handle is mAEffect, a member of the wrapper. Methods of VSTEffect do not operate on mAEffect directly, but through methods of VSTEffectWrapper. Is this wrong?